go/v1beta1/storage/filtering/types.go

```diff
@@ -7,16 +7,23 @@ type Query struct {
 }
 
 // Bool holds a general query that carries any number of
-// Must and Should operations
+// Must, MustNot, and Should operations
 type Bool struct {
-	Must   *Must   `json:"must,omitempty"`
-	Should *Should `json:"should,omitempty"`
-	Term   *Term   `json:"term,omitempty"`
+	Must    *Must    `json:"must,omitempty"`
+	MustNot *MustNot `json:"must_not,omitempty"`
+	Should  *Should  `json:"should,omitempty"`
+	Term    *Term    `json:"term,omitempty"`
 }
 
 // Must holds a must operator which each equates to an AND operation
 type Must []interface{}
 
+// MustNot holds a must_not operator which each equates to a != operation
+type MustNot struct {
+	Bool *Bool `json:"bool,omitempty"`
```

Comment:
is a must not that same thing as != or just !. If it is the former, then we won't need the Bool field here right?

Reply:
Like will it always be MustNot : { term : {left: right}}

Reply:
Answered in slack, but while I believe both are possible, `Bool` is used here so that more complicated queries can be nested within (using `&&` and `||` operators)